catalogue/admin.py

```diff
@@ -255,6 +256,15 @@ def get_readonly_fields(self, request, obj=None):
         else:
             return ("approved",)
 
+    def get_fieldsets(self, request, obj=None):
+        fieldsets = super(ComponentAdmin, self).get_fieldsets(request, obj)
+        admin_only = ["created_by"]
+        if not is_admin_or_mod(request):
+            fieldsets[0][1]['fields'] = tuple(x for x in fieldsets[0][1]['fields'] if x not in admin_only)
+        elif not all(x in fieldsets[0][1]['fields'] for x in admin_only):
+            fieldsets[0][1]['fields'] += tuple(admin_only)
```

Comment:
needs comment.

is the `elif` necessary, if `created_by` would already be defined in the fieldset? We would only have to hide the field for normal users.

Reply:
yes, since it gets removed again after reload. We would need to define the fieldsets hardcoded in the function and then only append "admin_only" if it's an admin. The problem is that after each reload or swap of user type the modified fieldsets are consistent. So if i load the data as admin the fieldset gets added and then if i swap to user it needs to get removed from the fieldsets. It could be simplified when no swap between user roles is possible in the admin.